### PR TITLE
MIME4J-332 -- improve handling of extended and continuation field name types

### PR DESCRIPTION
**Pull request by tballison**

### CODE REVIEW

File: core/src/main/java/org/apache/james/mime4j/util/MimeParameterMapping.java [hunk 1]
```diff
@@ -36,13 +41,41 @@ public String get(String name) {
...
+        //switch on whether there's a section in the key
+        if (sectionDelimiter < 0) {
+            handleNoSection(key, value);
+        } else {
+            handleSection(key, value);
+        }
+    }
+
+    private void handleSection(String key, String value) {
+        //if the key there was not sectioned, remove it.
```
Comment (chibenwa) [inline core/src/main/java/org/apache/james/mime4j/util/MimeParameterMapping.java:60]: Maybe we could add that `we conform to https://bugzilla.mozilla.org/show_bug.cgi?id=588781 as part of MIME4J-332` so that the decision on the edge case, which is not obvious actually gets documented in the code too ?

File: dom/src/test/java/org/apache/james/mime4j/field/contentdisposition/ContentDispositionTest.java [hunk 2]
```diff
@@ -19,7 +19,12 @@
 
 package org.apache.james.mime4j.field.contentdisposition;
 
+import static org.junit.Assert.assertEquals;
+
+import java.io.StringReader;
+
 import org.apache.james.mime4j.MimeException;
+import org.apache.james.mime4j.field.contentdisposition.parser.ContentDispositionParser;
```
[the remark this answers is not in the document]
Comment (tballison) [reply dom/src/test/java/org/apache/james/mime4j/field/contentdisposition/ContentDispositionTest.java]: Gah. Sorry. Left over from earlier tests. Good catch. Thank you. Removed.

**Review by chibenwa — APPROVED** [review]

**Comment by tballison** [comment]
This is much larger than I had hoped for, but I think it does a better job of distinguishing standard, extended and continuing field name types. I had conflated them in my initial patch.

Let me know what you think.

**Review by Arsnael — APPROVED** [review]
Hi @tballison, it looks great thank you so much for the improvement contribution!

**Review by quantranhong1999 — APPROVED** [review]
